cmd/argo/commands/root.go

```diff
@@ -128,10 +132,33 @@ If your server is behind an ingress with a path (you'll be running "argo server
 		cli.SetLogLevel(logLevel)
 		cmdutil.SetGLogLevel(glogLevel)
 		log.WithField("version", argo.GetVersion()).Debug("CLI version")
+		printVersionMismatchWarning(command)
 	}
 	command.PersistentFlags().StringVar(&logLevel, "loglevel", "info", "Set the logging level. One of: debug|info|warn|error")
 	command.PersistentFlags().IntVar(&glogLevel, "gloglevel", 0, "Set the glog logging level")
 	command.PersistentFlags().BoolVarP(&verbose, "verbose", "v", false, "Enabled verbose logging, i.e. --loglevel debug")
 
 	return command
 }
+
+// printVersionMismatchWarning logs a warning if the CLI version does not match the server version
+func printVersionMismatchWarning(command *cobra.Command) {
+	// if ARGO_SERVER isn't set there's no need to compare server and cli version
```

Comment:
```suggestion
	// if ARGO_SERVER isn't set there's no need to compare server and cli version
```

I'm not sure that this is necessarily accurate. Any usage of the CLI could have unexpected results if there is a version mismatch. For example, through the k8s API, the CLI could be using an outdated spec

Reply:
you're right. a check would definitely be useful in this case as well. do you know whether there's a way to get a version that's comparable with the cli version directly via api access? I haven't been able to find one but then again I am not very familiar with the codebase yet.

Reply:
While I agree with:
> Any usage of the CLI could have unexpected results if there is a version mismatch. 

Even the `version` command does not check the server version if `ARGO_SERVER` is not defined. I'm assuming is not that simple to determine the version in that case.

https://github.com/argoproj/argo-workflows/blob/main/cmd/argo/commands/version.go#L23

Reply:
My comment was about the "there's no need to compare" comment. That is inaccurate.
You may not be able to retrieve it without `ARGO_SERVER` set, but is nonetheless inaccurate to say "there is no need compare". It would be more apt to say "you can't compare". But the comment is a bit redundant at that point

Reply:
Great, agree.